src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataParser;
+import core.basesyntax.service.ReportCreator;
+import core.basesyntax.service.ReportReader;
+import core.basesyntax.service.ReportWriter;
+import core.basesyntax.service.impl.DataParserImpl;
+import core.basesyntax.service.impl.ReportCreatorImpl;
+import core.basesyntax.service.impl.ReportReaderImpl;
+import core.basesyntax.service.impl.ReportWriterImpl;
+import core.basesyntax.strategy.BalanceStrategy;
+import core.basesyntax.strategy.DecreaseStrategy;
+import core.basesyntax.strategy.FruitStrategy;
+import core.basesyntax.strategy.IncreaseStrategy;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String INPUT_PATHNAME = "src/main/resources/data.csv";
+    private static final String REPORT_PATHNAME = "src/main/resources/report.csv";
+    private static Map<Operation, OperationHandler> strategyMap;
+
+    public static void main(String[] args) {
+        ReportReader reportReader = new ReportReaderImpl();
+        List<String> readFile = reportReader.readFile(INPUT_PATHNAME);
+
+        DataParser dataParser = new DataParserImpl();
+        List<FruitTransaction> fruitTransactions = dataParser.parseAll(readFile);
+
+        FruitDao fruitDao = new FruitDaoImpl();
+        mapBuilder(fruitDao);
+        FruitStrategy fruitStrategy = new FruitStrategy(strategyMap);
+        fruitStrategy.processData(fruitTransactions);
+
+        ReportCreator reportCreator = new ReportCreatorImpl();
+        String report = reportCreator.createReport(fruitDao.getFruits());
+
+        ReportWriter reportWriter = new ReportWriterImpl();
+        reportWriter.writeReport(report, REPORT_PATHNAME);
```

Comment:
I am not sure is it necessary to separate create report and write file.
I mean if we have to write report to CSV we need comma. If we need to write to some other format i would mean we don't need commas. It might be done as private method ReportWriterToCsv implements ReportWriter.

Reply:
Not sure either)) Thought this would look better than in one class